### PR TITLE
docs-util: fix update OAS schema

### PR DESCRIPTION
**Pull request by shahednasser**
- Fix updating an OAS schema when the types are completely different, or when the length of allOf/anyOf/oneOf arrays are different.
- Added description for `additional_data` to knowledge base.

### CODE REVIEW

File: www/utils/packages/docs-generator/src/classes/kinds/oas.ts [hunk 1]
```diff
@@ -1744,23 +1744,33 @@ class OasKindGenerator extends FunctionKindGenerator {
...
-    if (oldSchemaObj!.type !== newSchemaObj?.type) {
-      oldSchemaObj!.type = newSchemaObj?.type
-    }
+    const oldSchemaType = this.inferOasSchemaType(oldSchemaObj)
+    const newSchemaType = this.inferOasSchemaType(newSchemaObj)
 
-    if (
-      oldSchemaObj!.description !== newSchemaObj?.description &&
-      oldSchemaObj!.description === SUMMARY_PLACEHOLDER
+    if (oldSchemaType !== newSchemaType && newSchemaType) {
```
Comment (carlos-r-l-rodrigues) [inline www/utils/packages/docs-generator/src/classes/kinds/oas.ts]:
```suggestion
    if (oldSchemaType !== newSchemaType && newSchemaObj) {
```

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

File: www/utils/packages/docs-generator/src/classes/kinds/oas.ts (hunk 1 above)
Comment (shahednasser) [reply www/utils/packages/docs-generator/src/classes/kinds/oas.ts]: Nice catch 😄

**Comment by shahednasser** [comment]
@olivermrbl can we get this one in?

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**